### PR TITLE
(FACT-1858) disable legacy facts mechanism

### PR DESCRIPTION
**Pull request by gimmyxd**

### CODE REVIEW

File: lib/inc/internal/ruby/module.hpp [hunk 1]
```diff
@@ -92,6 +92,22 @@ namespace facter { namespace ruby {
          */
         facter::facts::collection& facts();
 
+        /**
+         * Gets the hidden_facts associated wot the module.
```
Comment (oanatmaria) [inline lib/inc/internal/ruby/module.hpp]: with

File: lib/src/ruby/module.cc [hunk 2]
```diff
@@ -204,6 +205,12 @@ namespace facter { namespace ruby {
 
         initialize_search_paths(paths);
 
+        _hide_legacy = _blocklist.find("legacy") != _blocklist.end();
+        facts.each([&](string const& name, value const* val){
+            _hidden_facts[name] = val->hidden();
+            return true;
+        });
```
Comment (gimmyxd) [inline lib/src/ruby/module.cc]: remove this

File: lib/src/ruby/module.cc [hunk 3]
```diff
@@ -535,6 +549,9 @@ namespace facter { namespace ruby {
 
     VALUE module::ruby_value(VALUE self, VALUE name)
     {
+        // module* instance = from_self(self);
+        // instance->collection.add(ruby.to_string(name), )
+
```
Comment (oanatmaria) [inline lib/src/ruby/module.cc]: Remove this

**Comment by gimmyxd** [comment]
Jenkins please test all on aix61-POWERf,amazon6-64,amazon7-64,arista4-32,debian8-64,debian9-32,debian10-64,fedora31-64,osx1014-64,redhat5-64,centos6-64,centos6-32,centos8-64,redhat7-64,redhat7-POWER,redhat7-AARCH64,redhat8-64,redhatfips7-64,sles12-64,sles12-POWER,sles15-64,solaris10-64,solaris11-64,solaris114-64,ubuntu1404-64,ubuntu1604-POWER,ubuntu1804-64,windows10ent-64a

**Comment by gimmyxd** [comment]
pxp-agent failures due to wrong pcp_broker_ref on pr testing
re-kicked solaris for zones failures

File: lib/inc/facter/facts/collection.hpp [hunk 4]
```diff
@@ -129,6 +129,14 @@ namespace facter { namespace facts {
         // Visual Studio 12 still doesn't allow default for move assignment.
         collection& operator=(collection&& other);
 
+        /**
+         * Gets the blocklist associated with the collection.
+         * @return Returns set of blocked groups.
+         */
+        std::set<std::string> blocklist(){
```
Comment (ciprianbadescu) [inline lib/inc/facter/facts/collection.hpp]: header should contain only declaration  `std::set<std::string> blocklist() const;`, while corresponding cpp file should contain the definition.
also `const` whould be used if class is not changed

File: lib/inc/internal/ruby/module.hpp [hunk 5]
```diff
@@ -92,6 +92,14 @@ namespace facter { namespace ruby {
          */
         facter::facts::collection& facts();
 
+        /**
+         * Gets hide_legacy.
+         * @return Returns boolean.
+         */
+         bool hide_legacy(){
```
Comment (ciprianbadescu) [inline lib/inc/internal/ruby/module.hpp]: header should contain only declaration `bool hide_legacy() const;`, while corresponding cpp file should contain the definition.
also const should be used if class is not changed

File: acceptance/tests/options/config_file/blocklist_legacy_from_puppet_facts.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+      step "Agent #{agent}: create default config file" do
+        # create the directories
+        on(agent, "mkdir -p '#{facter_conf_default_dir}'")
+        create_remote_file(agent, facter_conf_default_path, <<-FILE)
+        facts : { blocklist : [ "legacy" ] }
+        FILE
+      end
+
+      step 'blocked legacy facts should not be resolved' do
```
Comment (oanatmaria) [inline acceptance/tests/options/config_file/blocklist_legacy_from_puppet_facts.rb]: "should not be returned/shown in output" (they are resolved but not shown)

File: acceptance/tests/options/config_file/blocklist_legacy_from_puppet_facts.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        # if the lagacy facts are blocked, the fact shows only once
+        on(agent, puppet('facts')) do |puppet_facts_output|
+          assert_equal(
+            1,
+            puppet_facts_output.stdout.scan(/architecture/).count,
```
Comment (oanatmaria) [inline acceptance/tests/options/config_file/blocklist_legacy_from_puppet_facts.rb]: should scan for a legacy fact like 'operatingsystem'

**Comment by gimmyxd** [comment]
Jenkins please test all on aix61-POWERf,amazon6-64,amazon7-64,arista4-32,debian8-64,debian9-32,debian10-64,fedora31-64,osx1014-64,redhat5-64,centos6-64,centos6-32,centos8-64,redhat7-64,redhat7-POWER,redhat7-AARCH64,redhat8-64,redhatfips7-64,sles12-64,sles12-POWER,sles15-64,solaris10-64,solaris11-64,solaris114-64,ubuntu1404-64,ubuntu1604-POWER,ubuntu1804-64,windows10ent-64a with servertests

**Pull request closed without merging**